extension/runtime/src/main/java/org/iris_events/runtime/connection/ConnectionFactoryProvider.java

```diff
@@ -51,15 +52,27 @@ private ConnectionFactory buildConnectionFactory(IrisConfig config) {
             connectionFactory.setHost(config.getHost());
             connectionFactory.setPort(port);
             connectionFactory.setVirtualHost(vhost);
+
+            connectionFactory.setAutomaticRecoveryEnabled(true);
+            connectionFactory.setRecoveryDelayHandler(this);
```

Comment:
This replaces the #connectWithResilience logic from above?

Reply:
Yes. It uses the RecoveryDelayHandler from com.rabbitmq.client. I tested this by locally shutting down rabbitmq broker for increasing intervals up to 30min and it recovered nicely. All the topology was recovered, connections, channels, etc.

Reply:
that is great! 🥇